staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -62,7 +62,14 @@ func newWithClock(authenticator authenticator.Token, cacheErrs bool, successTTL,
 		cacheErrs:     cacheErrs,
 		successTTL:    successTTL,
 		failureTTL:    failureTTL,
-		cache:         newStripedCache(32, fnvHashFunc, func() cache { return newSimpleCache(128, clock) }),
+		// Cache performance degrades noticably when the number of
+		// tokens in operation exceeds the size of the cache. It is
+		// cheap to make the cache big in the second dimension below,
+		// the memory is only consumed when that many tokens are being
```

Comment:
If we consider the case of short lived SA tokens from the token request API, the cache will grow to the max size before dropping old tokens since there is no go routine or similar to purge the cache.  I do not think that is immediately apparent from this comment.

That being said, even at the worse case I do not think this is an issue.

Reply:
How short-lived are such tokens? How many of them do we make?

Reply:
And under what conditions do we make them?

Reply:
They live 10 minutes by default, and are provisioned per-pod.